akka-persistence-typed/src/main/scala/akka/persistence/typed/PersistenceId.scala

```diff
@@ -4,6 +4,128 @@
 
 package akka.persistence.typed
 
+object PersistenceId {
+
+  /**
+   * Default separator character used for concatenating a `typeHint` with `entityId` to construct unique persistenceId.
+   * This must be same as in Lagom's `scaladsl.PersistentEntity`, for compatibility. No separator is used
+   * in Lagom's `javadsl.PersistentEntity` so for compatibility with that the `""` separator must be used instead.
+   */
+  val DefaultSeparator = "|"
+
+  /**
+   * Constructs a [[PersistenceId]] from the given `entityTypeHint` and `entityId` by
+   * concatenating them with `|` separator.
+   *
+   * Cluster Sharding is often used together with `EventSourcedBehavior` for the entities.
+   * The `PersistenceId` of the `EventSourcedBehavior` can typically be constructed with:
+   * {{{
+   * PersistenceId(entityContext.entityTypeKey.name, entityContext.entityId)
+   * }}}
+   *
+   * That format of the `PersistenceId` is not mandatory and only provided as a convenience of
+   * a "standardized" format.
+   *
+   * Another separator can be defined by using the `apply` that takes a `separator` parameter.
+   *
+   * The `|` separator is also used in Lagom's `scaladsl.PersistentEntity` but no separator is used
+   * in Lagom's `javadsl.PersistentEntity`. For compatibility with Lagom's `javadsl.PersistentEntity`
+   * you should use `""` as the separator.
+   *
+   * @throws IllegalArgumentException if the `entityTypeHint` or `entityId` contains `|`
+   */
+  def apply(entityTypeHint: String, entityId: String): PersistenceId =
+    apply(entityTypeHint, DefaultSeparator, entityId)
+
+  /**
+   * Constructs a [[PersistenceId]] from the given `entityTypeHint` and `entityId` by
+   * concatenating them with the `separator`.
+   *
+   * Cluster Sharding is often used together with `EventSourcedBehavior` for the entities.
+   * The `PersistenceId` of the `EventSourcedBehavior` can typically be constructed with:
+   * {{{
+   * PersistenceId(entityContext.entityTypeKey.name, entityContext.entityId)
+   * }}}
+   *
+   * That format of the `PersistenceId` is not mandatory and only provided as a convenience of
+   * a "standardized" format.
+   *
+   * The default separator `|` is used by the `apply` that doesn't take a `separator` parameter.
+   *
+   * The `|` separator is also used in Lagom's `scaladsl.PersistentEntity` but no separator is used
+   * in Lagom's `javadsl.PersistentEntity`. For compatibility with Lagom's `javadsl.PersistentEntity`
+   * you should use `""` as the separator.
+   *
+   * @throws IllegalArgumentException if the `entityTypeHint` or `entityId` contains `separator`
+   */
+  def apply(entityTypeHint: String, separator: String, entityId: String): PersistenceId = {
```

Comment:
Yes, let's do that since that is the normal convention of optional parameters.
My thinking was to place them in the same order as the format with the separator in the middle.